### DISCUSSION TITLE
Search through Teams Messages

### DISCUSSION THREAD

**Discussion by ThomasVVK** [microsoft-search/pnp-modern-search#2343] [Q&A]
Hi,

I am currently trying to create verticals in order to reproduce Bing Microsoft Search scope.
I have been able to add verticals to search through SharePoint documents, people and Outlook messages but I couldn't link any search results webpart to Teams Messages.

Is there any feature allowing me to do that? If not, would a custom development be possible to accomplish this? 

I am looking for any track/documentation that could guide me through this.

Thank you very much!

**Comment by wobba** [comment]
There is no public search API exposing Teams messages yet, so not possible,

**Comment by ThomasVVK** [comment]
Thank you!

**Reply by FranckyC** [reply]
@ThomasVVK actually you should be able to do this with the latest version ([4.8.0 beta](https://github.com/microsoft-search/pnp-modern-search/releases/tag/4.8.0)). See https://github.com/microsoft-search/pnp-modern-search/pull/2364 and  https://docs.microsoft.com/en-us/graph/search-concept-chat-messages

**Reply by wobba** [reply]
I stand corrected :)
